pkg/agent/route/route_linux.go

```diff
@@ -315,7 +357,7 @@ func (c *Client) syncIPTables() error {
 	return nil
 }
 
-func (c *Client) restoreIptablesData(podCIDR *net.IPNet, podIPSet string, snatMarkToIP map[uint32]net.IP) *bytes.Buffer {
+func (c *Client) restoreIptablesData(podCIDR *net.IPNet, podIPSet, nodePortIPSet string, serviceGWHairpinIP net.IP, snatMarkToIP map[uint32]net.IP) *bytes.Buffer {
```

Comment:
I think the name `serviceGWHairpinIP` is misleading. It has nothing to do with hairpin but just a virtual IP indicating this is NodePort traffic.

Reply:
How about just serviceVirtualIP?